shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -539,12 +541,13 @@ void ExposeAPIInMainWorld(const std::string& key,
     context_bridge::ObjectCache object_cache;
     v8::Context::Scope main_context_scope(main_context);
 
-    v8::MaybeLocal<v8::Object> maybe_proxy = CreateProxyForAPI(
-        api_object, isolated_context, main_context, &object_cache, false, 0);
+    v8::MaybeLocal<v8::Value> maybe_proxy = PassValueToOtherContext(
+        isolated_context, main_context, api, &object_cache, false, 0);
     if (maybe_proxy.IsEmpty())
       return;
     auto proxy = maybe_proxy.ToLocalChecked();
-    if (!DeepFreeze(proxy, main_context))
+    if (proxy->IsObject() && !proxy->IsTypedArray() &&
```

Comment:
```suggestion
    if ((IsPlainObject(proxy) || IsPlainArray(proxy)) &&
```

I think that's the check we want here

Reply:
Imo it should be ` proxy->IsObject() && !proxy->IsTypedArray() `. The check ` (IsPlainObject(proxy) || IsPlainArray(proxy) ` skips for example promises. Also it'll make direct and nested values behave inconsistently. Take a look at `DeepFreeze` implementation. https://github.com/electron/electron/blob/5be50eee09720c3ce8956e9ecdb2f0cf55f1f725/shell/renderer/api/electron_api_context_bridge.cc#L74

**proxy->IsObject() && !proxy->IsTypedArray()**
![image](https://user-images.githubusercontent.com/22342377/99656352-0f2f1100-2a5d-11eb-8e64-717b83016b65.png)

**(IsPlainObject(proxy) || IsPlainArray(proxy)**
![image](https://user-images.githubusercontent.com/22342377/99656401-1ce49680-2a5d-11eb-8e30-741696c122ef.png)